### PR TITLE
feat: Added python sample codes for cloud-iot-token-service generateAccessToken API

### PR DESCRIPTION
**Pull request by codepriestz**
Adding access token python samples

### CODE REVIEW

File: samples/api-client/accesstoken_example/accesstoken.py [hunk 1]
```diff
@@ -0,0 +1,174 @@
...
+def generate_gcp_token(
+    project_id, cloud_region, registry_id, device_id, scope, algorithm, certificate_file
+):
+    """Generate GCP access token."""
+    # [START iot_generate_gcp_token]
+    # project_id = 'YOUR_PROJECT_ID'
+    # cloud_region = 'us-central1'
+    # registry_id = 'your-registry-id'
+    # device_id = 'your-device-id'
+    # scope = 'scope1 scope2'
```
Comment (gcseh) [inline samples/api-client/accesstoken_example/accesstoken.py:54]: Add scope list link to comment: https://developers.google.com/identity/protocols/oauth2/scopes

File: samples/api-client/accesstoken_example/accesstoken.py [hunk 2]
```diff
@@ -0,0 +1,174 @@
+#!/usr/bin/env python
```
Comment (gcseh) [inline samples/api-client/accesstoken_example/accesstoken.py:1]: would be nice to have a demo flow with access token generation and pubsub publish+storage upload.

File: samples/api-client/accesstoken_example/accesstoken.py [hunk 3]
```diff
@@ -0,0 +1,177 @@
...
+        help="Which encryption algorithm to use to generate the JWT.",
+    )
+    parser.add_argument("--certificate_path", help="Path to private certificate.")
+    parser.add_argument(
+        "--cloud_region", default="us-central1", help="GCP cloud region"
+    )
+
+    parser.add_argument("--device_id", default=None, help="Device id.")
+    parser.add_argument(
+        "--scope", default=None, help="Scope for GCP token. Space delimited strings"
```
Comment (gcseh) [inline samples/api-client/accesstoken_example/accesstoken.py:131]: add `See the full list of scopes at: https://developers.google.com/identity/protocols/oauth2/scopes` in the help text

File: samples/api-client/accesstoken_example/accesstoken_test.py [hunk 4]
```diff
@@ -0,0 +1,280 @@
...
+device_pubsub_topic = "projects/{}/topics/{}".format(project_id, device_topic_id)
+test_topic_id = "test-pubsub-{}".format(uuid.uuid4())
+
+# This format is used in the `clean_up_registries()` below.
+registry_id = "test-registry-{}-{}".format(uuid.uuid4().hex, int(time.time()))
+
+# Generate gcp access token, use gcp access token to create pubsub
+
+
+def test_generate_gcp_jwt_token_pubsub():
```
Comment (gcseh) [inline samples/api-client/accesstoken_example/accesstoken_test.py:48]: same comments for moving tests to the accesstoken.py as "feature" with the same region tags. see https://github.com/googleapis/nodejs-iot/pull/502

**Review by gcseh — APPROVED** [review]

File: samples/api-client/accesstoken_example/README.rst.in [hunk 5]
```diff
@@ -0,0 +1,25 @@
+# This file is used to generate README.rst
+
+product:
+  name: Google Cloud IoT Token Api
+  short_name: Cloud IoT Token Api
+  url: https://cloud.google.com/iot/docs
+  description: >
+    # Google Cloud IoT Core Python Device Federated Authentication example
+    This sample app demonstrates the capabilites of Google Cloud IoT Core device federated authentication feature.
+    Devices authenticated to Cloud IoT Core can use the [Token Service](https://cloud.google.com/iot/alpha/docs/reference/cloudiottoken/rest) federated authentication to request [OAuth 2.0 access tokens](https://developers.google.com/identity/protocols/oauth2) in exchange for their [Cloud IoT Core JWTs](https://cloud.google.com/iot/docs/how-tos/credentials/jwts).
```
Comment (dandhlee) [inline samples/api-client/accesstoken_example/README.rst.in:10]: `[Token Service](https://cloud.google.com/iot/alpha/docs/reference/cloudiottoken/rest)`
This is not making a good use of the `rst` template style, see [this file](https://github.com/GoogleCloudPlatform/python-docs-samples/blob/927a5641395093a7a9b388ce7516d030d4c07e18/datastore/cloud-client/README.rst) on how to format the links better for `rst` templates.

**Review by dandhlee — APPROVED** [review]
LGTM, one minor comment below for `.rst` style, but fine to merge as-is.

File: samples/api-client/accesstoken_example/README.rst.in (hunk 5 above)
Comment (tseaver) [reply samples/api-client/accesstoken_example/README.rst.in:10]: Why are we checking in the `.in` file here?  That file is part of the template, not the rendered output.

Comment (gcseh) [reply samples/api-client/accesstoken_example/README.rst.in]: @dandhlee Thanks for the heads up, links fixed.
@tseaver The `.in` files are checked in every repo, I think it is useful to have them checked in for later times when the samples changed.

**Pull request merged**